### PR TITLE
Refactoring Additional object code to use wrappers

### PR DESCRIPTION
**Pull request by callen-sfdo**
Replaced direct use of Data_Import_Field_Mapping__mdt and Data_Import_Object_Mapping__mdt cmt objects and replaced them with BDI_FieldMapping and BDI_ObjectMapping wrappers.  

I also reorganized the logic in the BDI_MappingServiceAdvanced class to streamline the code.  

In addition, I changed how the object mapping classes were working with describe info to reduce the amount of unnecessary describe info that is retrieved.

# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by callen-sfdo** [comment]
**lurch: attach W-036842

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001CC8eQAG" target="_blank">W-036842</a>

File: src/classes/BDI_ObjectMapping.cls [hunk 1]
```diff
@@ -50,19 +50,36 @@ public with sharing class BDI_ObjectMapping {
     @AuraEnabled public String Relationship_Field;
     @AuraEnabled public String Relationship_To_Predecessor;
 
+    public BDI_ObjectMapping() {
```
Comment (ptewson-sfdo) [inline src/classes/BDI_ObjectMapping.cls]: Comment block for this constructor?

File: src/classes/BDI_ObjectMapping.cls [hunk 2]
```diff
@@ -50,19 +50,36 @@ public with sharing class BDI_ObjectMapping {
...
+        
+    }
+
     public BDI_ObjectMapping(Data_Import_Object_Mapping__mdt objectMapping) {
```
Comment (ptewson-sfdo) [inline src/classes/BDI_ObjectMapping.cls]: And a comment block for this constructor!

**Pull request merged**